### PR TITLE
Add samtools stats to QC steps

### PR DESCRIPTION
**Pull request by hassanfa**
### This PR:

Added:

* Samtools to quality control steps

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/snakemake_rules/quality_control/multiqc.rule [hunk 1]
```diff
@@ -53,6 +53,11 @@ else:
     multiqc_input.extend(expand(bam_dir + "{sample}.mosdepth.region.dist.txt", sample=config["samples"]))
     multiqc_input.extend(expand(bam_dir + "{sample}.mosdepth.summary.txt", sample=config["samples"]))
 
+    # samtools metrics
+    multiqc_input.extend(expand(bam_dir + "{sample}.samtools.flagstats.txt", sample=config["samples"]))
+    multiqc_input.extend(expand(bam_dir + "{sample}.samtools.idxstats.txt", sample=config["samples"]))
+    multiqc_input.extend(expand(bam_dir + "{sample}.samtools.stats.txt", sample=config["samples"]))
+
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/multiqc.rule:60]: How about minimizing to one single command?

```
   multiqc_input.extend(expand(bam_dir + "{sample}.samtools.{stats}.txt", sample=config["samples"], stats=['flagstats', 'idxstats', 'stats']))
```

File: BALSAMIC/snakemake_rules/quality_control/samtools.rule [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  output:
+    flagstats = bam_dir + "{sample}.samtools.flagstats.txt",
+    idxstats = bam_dir + "{sample}.samtools.idxstats.txt",
+    stats = bam_dir + "{sample}.samtools.stats.txt",
+  params:
+    tmpdir = tempfile.mkdtemp(prefix=tmp_dir),
+    conda = config["bioinfo_tools"].get("samtools")
+  threads: get_threads(cluster_config, "samtools")
+  singularity: Path(singularity_image, config["bioinfo_tools"].get("samtools") + ".sif").as_posix() 
+  benchmark:
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/samtools.rule:16]: snakemake rule order like here https://balsamic.readthedocs.io/en/latest/snakemake_etiquette.html

File: BALSAMIC/snakemake_rules/quality_control/samtools.rule [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    benchmark_dir + "{sample}.samtools_flagstats_stats_idxstats.tsv"
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/samtools.rule:17]:
```suggestion
    benchmark_dir + "samtools_flagstats_stats_idxstats_{sample}.tsv"
```

File: BALSAMIC/snakemake_rules/quality_control/multiqc.rule [hunk 4]
```diff
@@ -53,6 +53,11 @@ else:
...
     if config["umiworkflow"]:
         multiqc_input.extend(expand(umi_qc_dir + "{sample}.umi.collect_hsmetric", sample=config["samples"]))
```
Comment (ashwini06) [inline BALSAMIC/snakemake_rules/quality_control/multiqc.rule:63]: Do we need samtools stats for umi consensus bam files as well?

File: BALSAMIC/snakemake_rules/quality_control/multiqc.rule (hunk 1 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/quality_control/multiqc.rule:60]: Good point. Agreed.

File: BALSAMIC/snakemake_rules/quality_control/multiqc.rule (hunk 4 above)
Comment (hassanfa) [reply BALSAMIC/snakemake_rules/quality_control/multiqc.rule:63]: I'll add that as well, why not.

**Comment by hassanfa** [comment]
UMI and WGS require Sentieon as container or two independent rules for samtools. Maybe for another PR. Merging.

**Pull request merged**